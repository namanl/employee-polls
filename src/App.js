import { connect } from 'react-redux';
import { useEffect } from 'react';
import { handleInitialData } from './actions/shared';
import { Route, Routes, Navigate, useLocation } from 'react-router-dom';

import Login from './components/Login';
import Dashboard from './components/Dashboard';
import PollPage from './components/PollPage';
import NewPoll from './components/NewPoll';
import Nav from './components/Nav';
import LeaderBoard from './components/LeaderBoard';
import PageNotFound from './components/PageNotFound';

function App(props) {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, [props]);

  const location = useLocation();
  const authed = props.authedUser !== null;

  function ProtectedRoute({ children }) {
    return authed ? (
      children
    ) : (
      <Navigate to="/login" replace state={{ path: location.pathname }} />
    );
  }

  return (
    <div className="container mx-auto py-4">
      {!authed ? null : <Nav />}
      <Routes>
        <Route
          path="/"
          exact
          element={
            <ProtectedRoute>
              <Dashboard />
            </ProtectedRoute>
          }
        />
        <Route
          path="/new"
          exact
          element={
            <ProtectedRoute>
              <NewPoll />
            </ProtectedRoute>
          }
        />
        <Route
          path="/questions/:id"
          exact
          element={
            <ProtectedRoute>
              <PollPage />
            </ProtectedRoute>
          }
        />
        <Route
          path="/leaderboard"
          exact
          element={
            <ProtectedRoute>
              <LeaderBoard />
            </ProtectedRoute>
          }
        />
        <Route path="/login" element={<Login />} />
        <Route path="*" element={<PageNotFound />} />
      </Routes>
    </div>
  );
}

const mapStateToProps = ({ authedUser }) => ({
  authedUser,
});

export default connect(mapStateToProps)(App);
