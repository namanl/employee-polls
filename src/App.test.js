import { render } from '@testing-library/react';
import * as React from 'react';
import App from './App';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore } from 'redux';
import reducer from './reducers';
import { applyMiddleware } from 'redux';
import { MemoryRouter } from 'react-router';

const store = createStore(reducer, applyMiddleware(thunk));

describe('App', () => {
  it('matches the snapshot', () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
