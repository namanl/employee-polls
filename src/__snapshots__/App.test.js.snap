// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto py-4"
      >
        <div>
          <h1
            class="m-4 text-center font-bold text-3xl"
            data-test-id="login-heading"
          >
            Employee Polls
          </h1>
          <img
            alt="Employees"
            class="block mt-0 mb-0 ml-auto mr-auto w-[30%]"
            src="images.jpeg"
          />
          <h2
            class="text-center m-4 font-bold text-2xl"
            data-testid="login-heading"
          >
            Login
          </h2>
          <form>
            <div
              class="m-2"
            >
              <label
                class="p-2 block text-center text-sm font-medium text-slate-700"
                for="user"
              >
                User
              </label>
              <input
                class="block w-full border border-gray-300 rounded-md p-2"
                data-testid="username"
                id="username"
                name="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="m-2"
            >
              <label
                class="p-2 block text-center text-sm font-medium text-slate-700"
                for="password"
              >
                Password
              </label>
              <input
                class="block w-full border border-gray-300 rounded-md p-2"
                data-testid="password"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="m-6 text-center"
            >
              <button
                class="bg-gray-300 cursor-not-allowed  px-5 py-2.5 text-sm leading-5 rounded-md font-semibold text-white"
                data-testid="submit"
                disabled=""
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto py-4"
    >
      <div>
        <h1
          class="m-4 text-center font-bold text-3xl"
          data-test-id="login-heading"
        >
          Employee Polls
        </h1>
        <img
          alt="Employees"
          class="block mt-0 mb-0 ml-auto mr-auto w-[30%]"
          src="images.jpeg"
        />
        <h2
          class="text-center m-4 font-bold text-2xl"
          data-testid="login-heading"
        >
          Login
        </h2>
        <form>
          <div
            class="m-2"
          >
            <label
              class="p-2 block text-center text-sm font-medium text-slate-700"
              for="user"
            >
              User
            </label>
            <input
              class="block w-full border border-gray-300 rounded-md p-2"
              data-testid="username"
              id="username"
              name="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="m-2"
          >
            <label
              class="p-2 block text-center text-sm font-medium text-slate-700"
              for="password"
            >
              Password
            </label>
            <input
              class="block w-full border border-gray-300 rounded-md p-2"
              data-testid="password"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="m-6 text-center"
          >
            <button
              class="bg-gray-300 cursor-not-allowed  px-5 py-2.5 text-sm leading-5 rounded-md font-semibold text-white"
              data-testid="submit"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
